src/PubSub/Drivers/RedisClient.php

```diff
@@ -209,7 +248,7 @@ public function joinChannel($appId, string $channel, string $socketId, string $d
      */
     public function leaveChannel($appId, string $channel, string $socketId)
     {
-        $this->publishClient->__call('hdel', [$this->getTopicName($appId, $channel), $socketId]);
+        $this->redis->hdel($this->getTopicName($appId, $channel), $socketId);
```

Comment:
I don't agree with these changes as I said above, this is making this synchronous and blocking rather than async and non-blocking.